weave/trace/weave_client.py

```diff
@@ -641,8 +654,10 @@ def create_call(
         attributes._set_weave_item("os_release", platform.release())
         attributes._set_weave_item("sys_version", sys.version)
 
+        op_name_future = self.future_executor.defer(lambda: op_def_ref.uri())
```

Comment:
Can you help me understand how this works?

It seems like we:
1. Save an op, which generates a ref
2. That ref's URI is not necessarily available yet, so we make a future of it and pass it as the value of `call._op_name`
3. The `call_start` is also deferred (but we don't wait for the uri to resolve first?)
4. The `end_call` is also deferred (also not clear if there are guarantees this always comes after call_start -- what if you start and stop immediately?)